src/App.tsx

```diff
@@ -15,47 +16,116 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  ByAlphabet = 'byAlphabet',
+  ByLength = 'byLength',
+  NoSort = '',
+}
+
+const prepeareGoods = (
+  goods: string[],
+  sortType: SortType,
+  isReversed: boolean,
+) => {
+  const goodsToEdit = [...goods];
+
+  if (sortType) {
+    goodsToEdit.sort((firstGood, secondGood) => {
+      switch (sortType) {
+        case SortType.ByAlphabet:
+          return firstGood.localeCompare(secondGood);
+        case SortType.ByLength:
+          return firstGood.length - secondGood.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    goodsToEdit.reverse();
+  }
+
+  return goodsToEdit;
+};
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState(SortType.NoSort);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const setIsReversedToggle = () => {
+    if (isReversed === true) {
+      setIsReversed(false);
+    } else {
+      setIsReversed(true);
+    }
```

Comment:
You can just toggle this value depending on previous one, with functional setState
`setIsReversed(prev => !prev)`